File: sub-web/src/test/java/org/github/MyBatisGenerator.java
package org.github;

import lombok.val;

import com.baomidou.mybatisplus.generator.AutoGenerator;
import com.baomidou.mybatisplus.generator.config.DataSourceConfig;
import com.baomidou.mybatisplus.generator.config.GlobalConfig;
import com.baomidou.mybatisplus.generator.config.PackageConfig;
import com.baomidou.mybatisplus.generator.config.StrategyConfig;
import com.baomidou.mybatisplus.generator.engine.FreemarkerTemplateEngine;

import static com.baomidou.mybatisplus.annotation.IdType.ID_WORKER_STR;
import static com.baomidou.mybatisplus.generator.config.rules.NamingStrategy.underline_to_camel;
import static java.lang.System.getProperty;

public class MyBatisGenerator {
  public static void main(String[] args) {
    val path         = getProperty("user.dir");
    val globalConfig = new GlobalConfig();
    globalConfig
      .setOutputDir(path + "/sub-web/src/main/java")
      .setFileOverride(true)
      .setOpen(false)
      .setAuthor("JYD_XL")
      .setBaseResultMap(true)
      .setIdType(ID_WORKER_STR)
      .setEntityName("%sEntity")
      .setMapperName("%sMapper")
      .setXmlName("%sMapper")
      .setServiceName("I%sService")
      .setServiceImplName("%sServiceImpl")
      .setControllerName("%sController");

    val dataSourceConfig = new DataSourceConfig();
    dataSourceConfig
      .setUrl("jdbc:mysql://localhost:3357/shiro?useSSL=false")
      .setDriverName("com.mysql.jdbc.Driver")
      .setUsername("root")
      .setPassword("l");

    val packageConfig = new PackageConfig();
    packageConfig
      .setParent("org.github")
      .setXml("mapper.base")
      .setModuleName("base");

    val strategyConfig = new StrategyConfig();
    strategyConfig
      .setSkipView(true)
      .setNaming(underline_to_camel)
      .setEntityLombokModel(true)
      .setRestControllerStyle(true)
      .setControllerMappingHyphenStyle(true)
      .setInclude(".+");

    val autoGenerator = new AutoGenerator();
    autoGenerator
      .setGlobalConfig(globalConfig)
      .setDataSource(dataSourceConfig)
      .setPackageInfo(packageConfig)
      .setStrategy(strategyConfig)
      .setTemplateEngine(new FreemarkerTemplateEngine())
      .execute();
  }
}

File: sub-model/src/test/java/org/github/MyBatisGenerator.kt
package org.github

import com.baomidou.mybatisplus.annotation.IdType.*
import com.baomidou.mybatisplus.generator.AutoGenerator
import com.baomidou.mybatisplus.generator.config.DataSourceConfig
import com.baomidou.mybatisplus.generator.config.GlobalConfig
import com.baomidou.mybatisplus.generator.config.PackageConfig
import com.baomidou.mybatisplus.generator.config.StrategyConfig
import com.baomidou.mybatisplus.generator.config.TemplateConfig
import com.baomidou.mybatisplus.generator.config.po.LikeTable
import com.baomidou.mybatisplus.generator.config.rules.NamingStrategy.*
import com.baomidou.mybatisplus.generator.engine.FreemarkerTemplateEngine
import org.github.base.AbstractEntity
import java.lang.System.*

fun main() {
  val generator = AutoGenerator().apply { templateEngine = FreemarkerTemplateEngine() }

  GlobalConfig().apply {
    val path = checkNotNull(getProperty("user.dir"))
    outputDir = "$path/sub-model/src/main/java"
    serviceImplName = "%sServiceImpl"
    serviceName = "I%sService"
    mapperName = "I%sMapper"
    entityName = "%sEntity"
    isFileOverride = true
    isOpen = false
    author = "JYD_XL"
    idType = ASSIGN_ID
    generator.globalConfig = this
  }

  DataSourceConfig().apply {
    url = "jdbc:mysql://ubuntu:3380/shop"
    driverName = "com.mysql.cj.jdbc.Driver"
    username = "root"
    password = "root"
    generator.dataSource = this
  }

  PackageConfig().apply {
    parent = "org.github"
    moduleName = "base"
    generator.packageInfo = this
  }

  StrategyConfig().apply {
    naming = underline_to_camel
    isEntityLombokModel = true
    isEntityTableFieldAnnotationEnable = true
    logicDeleteFieldName = "deleted"
    setSuperEntityClass(AbstractEntity::class.java)
    superMapperClass = "org.github.base.IMapper"
    superServiceClass = "org.github.base.IService"
    superServiceImplClass = "org.github.base.ServiceImpl"
    likeTable = LikeTable("_")
    generator.strategy = this
  }

  TemplateConfig().apply {
    controller = null
    xml = null
    service = "service.java"
    serviceImpl = "serviceImpl.java"
    mapper = "mapper.java"
    setEntity("entity.java")
    generator.template = this
  }

  generator.execute()
}
